tests/common/test_protobuf_validation.nim

```diff
@@ -0,0 +1,103 @@
+
```

Comment:
nice. I actually was thinkinig about testing the `encode`, `decode` functions of the protos that were changed, like `PagingInfoRPC`.

A simple back and forth converstion like [this](https://github.com/status-im/nimbus-eth2/blob/5b13b7cc9bd52617f81c2b0ce5eae06827741404/tests/test_eth2_ssz_serialization.nim#L28-L30) to assert that encode()/decode() work as expected and protect the code from future modifications. Also, test this extra logic `if topic.len > MaxPubsubTopicLen:` trying to encode eg a `FilterRequest` with a longer topic, and asserting it fails.

But we can leave that for another PR.

Reply:
(ofc testing all encode/decode functions would be a long PR) beyond this, but perhaps we are lacking some testing in protos in general?

Reply:
> (ofc testing all encode/decode functions would be a long PR) beyond this, but perhaps we are lacking some testing in protos in general?

Protocol-specific protobuf tests can be found in the different protocol folders (see the `tests/v2/waku_store` module). Additionally, the different protocols' integration/end-to-end tests implicitly test protocol RPC serialisation. 

I don't see the need to add test suites for unit testing/covering specifically the RPC serialization of the different protocols. If we had to add tests to cover this area: Interoperability tests (a.k.a integration-tests) would be the ones providing additional value.